### PR TITLE
ratpoints: 2.1.3 -> 2.1.3.p4; add darwin support

### PR DESCRIPTION
**Pull request by jbaum98**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by adisbladis** [comment]
@GrahamcOfBorg build ratpoints

File: pkgs/applications/science/math/ratpoints/default.nix [hunk 1]
```diff
@@ -1,20 +1,34 @@
...
   src = fetchurl {
     url = "http://www.mathe2.uni-bayreuth.de/stoll/programs/ratpoints-${version}.tar.gz";
     sha256 = "0zhad84sfds7izyksbqjmwpfw4rvyqk63yzdjd3ysd32zss5bgf4";
   };
-  buildInputs = [gmp];
-  makeFlags = "INSTALL_DIR=$(out)";
+
+  patches = [
+    (fetchpatch {
+      url = "https://git.sagemath.org/sage.git/plain/build/pkgs/ratpoints/patches/sturm_and_rp_private.patch";
```
Comment (Mic92) [inline pkgs/applications/science/math/ratpoints/default.nix]: Can your refer to a specific commit? Otherwise an update of the patch will break the hash.

**Comment by mpickering** [comment]
Looks good to me now the issue @Mic92 pointed out has been fixed.

**Comment by adisbladis** [comment]
Thanks :)

**Pull request merged**